tsql/tsql.g4

```diff
@@ -217,12 +219,42 @@ create_index
     ;
 
 // https://msdn.microsoft.com/en-us/library/ms187926(v=sql.120).aspx
-create_procedure
-    : CREATE proc=(PROC | PROCEDURE) func_proc_name (';' DECIMAL)?
+create_alter_procedure
+    : (CREATE | ALTER) proc=(PROC | PROCEDURE) func_proc_name (';' DECIMAL)?
       ('('? procedure_param (',' procedure_param)* ')'?)?
       (WITH procedure_option (',' procedure_option)*)?
       (FOR REPLICATION)? AS sql_clauses
     ;
+    
+//https://msdn.microsoft.com/en-us/library/ms186755.aspx
+create_alter_function
+    : (CREATE | ALTER) FUNCTION func_proc_name
+        (('(' procedure_param (',' procedure_param)* ')') | '(' ')') //must have (), but can be empty
+        (function_type1 | function_type2 | function_type3) ';'?
+
+    ;
+
+function_type1:RETURNS TABLE
+               (WITH function_option (',' function_option)*)?
+               AS?
+               RETURN select_statement ;
+
+function_type2: RETURNS LOCAL_ID table_type_definition
+                (WITH function_option (',' function_option)*)?
+                AS?
+                BEGIN
+                   sql_clause*
+                   RETURN
+                END ;
+
+
+function_type3:RETURNS data_type
```

Comment:
How about `inline_table_valued_function`, `multistatement_table_valued_function` and `scalar_function` instead of `function_type1`, `function_type2` and `funcation_type3`?